### PR TITLE
Feature/MOB-4198   quote list   use vehicle images from be

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
## JIRA
[MOB-4198](https://karhoo.atlassian.net/browse/MOB-4198)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooSDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4314,7 +4326,7 @@
 				CODE_SIGN_IDENTITY = "";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 28;
+				CURRENT_PROJECT_VERSION = 29;
```
Comment (dianakarh) [inline KarhooSDK.xcodeproj/project.pbxproj:4329]: Did you bump the version in preparation for the release? If so, we'll need to remember not to bump again in 2 week's time. I'd rather bump the version then, or at least let the team know this has been changed already

File: KarhooSDK/Network/HttpClient/JsonHttpClient/JsonHttpClient.swift [hunk 2]
```diff
@@ -131,6 +133,13 @@ private extension JsonHttpClient {
     func absoluteUrl(endpoint: APIEndpoint) -> URL {
         let environmentDetails = KarhooEnvironmentDetails(environment: Karhoo.configuration.environment())
 
+        guard endpoint != .vehicleRules else {
```
Comment (dianakarh) [inline KarhooSDK/Network/HttpClient/JsonHttpClient/JsonHttpClient.swift:136]: let's have this guard before the environment details retrieval in order to provide a better separation

File: KarhooSDK/Service/Quote/QuoteService.swift [hunk 3]
```diff
@@ -12,4 +12,5 @@ public protocol QuoteService {
     func quotes(quoteSearch: QuoteSearch) -> PollCall<Quotes>
     func coverage(coverageRequest: QuoteCoverageRequest) -> Call<QuoteCoverage>
     func verifyQuote(verifyQuotePayload: VerifyQuotePayload) -> Call<Quote>
+    func getVehiclesRules() -> Call<VehicleRules>
```
Comment (dianakarh) [inline KarhooSDK/Service/Quote/QuoteService.swift:15]: I think `getVehicleImageRules` would be more descriptive. The rules are specific to the vehicle image

Comment (AleksanderWedrychowskiKarhoo) [reply KarhooSDK/Service/Quote/QuoteService.swift]: 👍  changed all names to include `Image` word.

File: KarhooSDK.xcodeproj/project.pbxproj (hunk 1 above)
Comment (AleksanderWedrychowskiKarhoo) [reply KarhooSDK.xcodeproj/project.pbxproj]: In last commit I reverted those changes, so we keep release process untouched.

**Review by BartlomiejSopalaKarhoo — DISMISSED** [review]

File: KarhooSDK/Api/Response/VehicleImageRules.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        case rules = "mappings"
+    }
+}
+
+public struct VehicleImageRule: Codable {
+    
+    private static let fallbackRuleValue = "*"
+
+    public enum RuleType {
+        /// Rule that should be used in a first place. Vehicle should mach type and it's tags list should be same as tags enlisted in a rule.
```
Comment (dianakarh) [inline KarhooSDK/Api/Response/VehicleImageRules.swift:24]: you have some typos in the description. took the opportunity to rephrase it for clarity: `Rule that should be used when the vehicle type and tags match perfectly the ones in the quote.`

File: KarhooSDK/Api/Response/VehicleImageRules.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        case specific
+        /// Rule default for given vehicle type. If vehicle does not required tags, this rule should be used.
```
Comment (dianakarh) [inline KarhooSDK/Api/Response/VehicleImageRules.swift:26]: Default rule for a given vehicle type. If the quote does not contain vehicle tags, then this rule should be used.*

**Review by BartlomiejSopalaKarhoo — APPROVED** [review]

**Review by dianakarh — APPROVED** [review]

**Pull request merged**